src/pal/src/map/map.cpp

```diff
@@ -2241,7 +2241,9 @@ MAPmmapAndRecord(
     PAL_ERROR palError = NO_ERROR;
     LPVOID pvBaseAddress = NULL;
 
-    pvBaseAddress = mmap(addr, len, prot, flags, fd, offset);
+    off_t adjust = offset & (VIRTUAL_PAGE_MASK);
+
+    pvBaseAddress = mmap(static_cast<char *>(addr) - adjust, len + adjust, prot, flags, fd, offset - adjust);
```

Comment:
mmap for MAP_PRIVATE documentation is unclear about whether two PRIVATE mappings of the same file affect each other.

```
MAP_PRIVATE
        Create a private copy-on-write mapping.  Updates to the mapping are not 
        visible to other  processes  mapping  the same file, and are not carried 
        through to the underlying file.  It is unspecified whether changes made to 
        the file after the mmap() call are visible in the mapped region.
```

For the amd64 issue, the mmap documentation is sufficiently clear.
```
MAP_FIXED
          Don't interpret addr as a hint: place the mapping at exactly that 
          address.  addr must be a multiple  of  the  page size.   If  the  
          memory region specified by addr and len overlaps pages of any 
          existing mapping(s), then the overlapped part of the existing 
          mapping(s) will be discarded.  If the specified address cannot be  
          used,  mmap()  will fail.  Because requiring a fixed address for a 
          mapping is less portable, the use of this option is discouraged.
```